wordpress-mobile/WordPress-Android
# Add Jetpack branding to Scan screens starting from Phase One

Fixes Partially #17334

This PR adds the logic to display the Jetpack powered branding banner on the Scan screens starting from Phase One of features removal.

## To test:

Jetpack branding on Scan screens should be visible starting from phase one of features removal, while it should be hidden when all Jetpack removal flags are disabled.

Normal Phase - Jetpack Branding not shown
- This is the phase when the `Jetpack powered` should **not** be displayed
- Enable this phase by making sure all Jetpack removal flags are disabled

Branded Phase - Jetpack Branding shown
- This is the phase where we show a `Jetpack powered` branding on the Stats screens
- Enable this phase by making sure `jp_removal_one` flag is enabled
- Go to `Me` → `App Settings` → `Debug settings` and **enable** `jp_removal_one`
- Additionally, any other Jetpack removal phase flag should end up in the branding being shown 

Test the following steps in each phase:

#### Testing Steps

1. Open WordPress app and login with site address, ❗ the site should have Jetpack Scan _Daily_ enabled
   1. Additionally, login with your WP.com account and make sure Jetpack is connected to your WP.com account
   2. Make sure the selected site is the self-hosted one, from step `1`
2. Toggle one of the phases based on the notes above testing steps
3. Go to `My Site` → `Menu` tab
4. From menu, under `Jetpack` tap `Scan`
5. If in normal phase, expect no `Jetpack powered` branding
6. If in branded phase:
   1. **Expect** the `Jetpack powered` banner to be visible at the bottom of the screen
   2. Tap the `Jetpack powered` banner
   3. **Expect** the `WordPress is better with Jetpack` bottom sheet to be displayed
   4. **Optionally** tap the `Try the new Jetpack app` button
      1. If Jetpack app is not installed it should open the Play Store to install the app
      2. If Jetpack app is installed, it should be opened
   5. Go back to the WordPress app
7. Tap `History` at the top right of the Scan screen
8. If in normal phase, expect no `Jetpack powered` branding
9. If in branded phase:
   1. **Expect** the `Jetpack powered` banner to be visible at the bottom of the screen
   2. Tap the `Jetpack powered` banner
   3. **Expect** the `WordPress is better with Jetpack` bottom sheet to be displayed
   4. **Optionally** tap the `Try the new Jetpack app` button
      1. If Jetpack app is not installed it should open the Play Store to install the app
      2. If Jetpack app is installed, it should be opened

### Screenshots

> **Note** I've used dummy data for the screenshots to make the Scan screens scrollable.

| Jetpack Scan | Jetpack Scan History |
| - | - |
| <img width="314" src="https://user-images.githubusercontent.com/4588074/207313857-11af481c-cf22-43ca-900e-9cb9d963cdee.png"> | <img width="314" src="https://user-images.githubusercontent.com/4588074/207313881-aa276063-908c-4879-97a9-9e8b3d71d73e.png"> |

## Regression Notes
1. Potential unintended areas of impact
   Scan and Scan history screen.

2. What I did to test those areas of impact (or what existing automated tests I relied on)
   Manual testing

3. What automated tests I added (or what prevented me from doing so)
   - Added tests in `JetpackFeatureRemovalBrandingUtilTest`

PR submission checklist:
